Plugins/Json/src/DetectorJsonConverter.cpp

```diff
@@ -108,32 +122,55 @@ nlohmann::json Acts::DetectorJsonConverter::toJsonDetray(
       nSurfaces += jVolume["surfaces"].size();
     }
   }
-  jData["volumes"] = jVolumes;
-
-  /// @TODO fix this to actual volume grid
-  nlohmann::json jVolumeGridParent;
-  nlohmann::json jVolumeGrid;
-  nlohmann::json jVolumeGridAxes;
-  jVolumeGrid["axes"] = jVolumeGridAxes;
-  jVolumeGrid["entries"] = std::vector<std::size_t>{};
-  jVolumeGridParent["grid"] = jVolumeGrid;
-  jData["volume_grid"] = jVolumeGridParent;
-
-  // Get the time stamp
-  // Write the header
-  nlohmann::json jHeader;
-  jHeader["detector"] = detector.name();
-  jHeader["date"] = std::asctime(ti);
-  jHeader["volume_count"] = detector.volumes().size();
+  jGeometryData["volumes"] = jVolumes;
+  jGeometryData["volume_grid"] = DetectorVolumeFinderJsonConverter::toJson(
+      detector.detectorVolumeFinder(), true);
 
+  // (1) Geometry
   // For detray, the number of surfaces and portals are collected
   // from the translated volumes
-  jHeader["type"] = "detray";
-  jHeader["surface_count"] = nSurfaces;
-  jDetector["header"] = jHeader;
-  jDetector["data"] = jData;
+  jGeometryHeader["type"] = "detray";
```

Comment:
Should be clear from the `version` field by now

Reply:
yes, this is more for the ACTS side.